x-pack/plugins/search_playground/server/routes.ts

```diff
@@ -45,22 +61,42 @@ export function defineRoutes({ log, router }: { log: Logger; router: IRouter })
       path: APIRoutes.POST_CHAT_MESSAGE,
       validate: {
         body: schema.object({
-          data: schema.any(),
+          data: schema.object({
+            connector_id: schema.string(),
+            indices: schema.string(),
+            prompt: schema.string(),
+            citations: schema.boolean(),
+            elasticsearch_query: schema.string(),
+            summarization_model: schema.string(),
+            doc_size: schema.number(),
+            source_fields: schema.string(),
+          }),
           messages: schema.any(),
         }),
       },
     },
     errorHandler(async (context, request, response) => {
+      const [_, { encryptedSavedObjects }] = await getStartServices();
       const { client } = (await context.core).elasticsearch;
+      const encryptedSavedObjectsClient = encryptedSavedObjects.getClient({
+        includedHiddenTypes: ['action'],
+      });
 
       const aiClient = Assist({
         es_client: client.asCurrentUser,
       } as AssistClientOptionsWithClient);
 
       const { messages, data } = await request.body;
 
+      const decryptedApiKey =
+        await encryptedSavedObjectsClient.getDecryptedAsInternalUser<RawAction>(
+          'action',
+          data.connector_id
+        );
+
       const model = new ChatOpenAI({
-        openAIApiKey: data.api_key,
+        modelName: data.summarization_model,
```

Comment:
you could create a subaction in the connector that returns this model

Reply:
However, you need to consider token tracking so it may be better to extend the `ChatOpenAI` class and [override the _call method like we are doing in security solution with LLM](https://github.com/elastic/kibana/blob/a85ba2b7ee276ffd99e86f1d5c6622b53b573ea2/x-pack/plugins/elastic_assistant/server/lib/langchain/llm/actions_client_llm.ts#L68).

Otherwise you need to implement token tracking for your new subaction. This would be done in [getGenAITokenTracking](https://github.com/elastic/kibana/blob/d62490699b56c287d2e264129e00580bb8668302/x-pack/plugins/actions/server/lib/gen_ai_token_tracking.ts#L30). It is [called during the event log process](https://github.com/elastic/kibana/blob/d62490699b56c287d2e264129e00580bb8668302/x-pack/plugins/actions/server/lib/action_executor.ts#L573).

Reply:
Yeah, we are using 'default model' defined in connector as a default value for models, but according to design we should able to easily change it, without changing connector model